guava/src/com/google/common/math/Quantiles.java

```diff
@@ -603,22 +639,30 @@ private static void movePivotToStartOfSlice(double[] array, int from, int to) {
    * allRequired[i]} for {@code i} in the range [{@code requiredFrom}, {@code requiredTo}]. These
    * indexes must be sorted in the array and must all be in the range [{@code from}, {@code to}].
    */
+  @SuppressWarnings({"lowerbound:argument.type.incompatible",/*(1): Since allRequired is sorted, and requiredBelow < requiredChosen,
+  allRequired[requiredBelow] <= allRequired[requiredChosen], therefore allRequired[requiredBelow] is always <= required. Based on the condition
+  of if and while, allRequired[requiredBelow] != required. Because allRequired[requiredBelow] <= required, we know that allRequired[requiredBelow] < required,
+  since allRequired[requiredBelow] >= requiredFrom >= 0, then required > 0. */
+          "upperbound:argument.type.incompatible"/*(2): Since allRequired is sorted, and requiredAbove > requiredChosen,
+  allRequired[requiredAbove] >= allRequired[requiredChosen], therefore allRequired[requiredAbove] is always >= required. Based on the condition
+  of if and while, allRequired[requiredAbove] != required. Because allRequired[requiredAbove] >= required, we know that allRequired[requiredAbove] > required,
+  since requiredTo >= allRequired[requiredAbove] > required, required + 1 < array.length. */})
```

Comment:
Again, I don't see why `requiredTo >= allRequired[requiredAbove]` should hold. Can you write `array.length > allRequired[requiredAbove]` instead?
Otherwise, the explanation looks good.